### PR TITLE
Improve SHT3xD reconnect handling

### LINKED ISSUE #550: status_led does not light up on a failing i2c sensor
**Operating environment/Installation (Hass.io/Docker/pip/etc.):** HA installed in vmware

**ESP (ESP32/ESP8266, Board/Sonoff):** ESP32 WROOM 32D

The ESP32 is mounted on a custom breakout board with the required pullup resistors, capacitors and 3.3V LDO (NCP114)

**Affected component:** Status LED

https://esphome.io/components/status_led.htm

**Description of problem:**
The **sensor_led** component which is supposed to blink on warning or error events risen from other components does not do so when an i2c sensor is diconnected.

The DUT is an ESP32D with a red LED wired to IO12 through a 10K resistor.

During the bootup, or if the wifi gets diconnected, the LED blinks as expected until the connection is reestablished., so the wiring and the basic functionality of the Status LED appears to be as expected.

But if the i2c sensors are disconnected from the board, the Status LED does not begin to blink, eventhough the i2c component starts throwing following WARNING events:
`[16:42:03][W][i2c:070]: Received NACK on transmit of address 0x4B`

**Problem-relevant YAML-configuration entries:**
```yaml
esphome:
  name: sensornode
  platform: ESP32
  board: nodemcu-32s

wifi:
  ssid: "ssid"
  password: "passwd"

# Enable logging
logger:

# Enable Home Assistant API
api:
  password: "passwd"

ota:
  password: "passwd"
i2c:
  sda: 21
  scl: 22
  scan: True
  frequency: 50kHz  
sensor:
  - platform: wifi_signal
    name: "sensornode wifi signal sensor"
    update_interval: 60s
  - platform: uptime
    name: "sensornode uptime sensor"
    update_interval: 60s
  - platform: sht3xd
    temperature:
      name: "sensornode SHT30 0x44 Temperature"
      accuracy_decimals: 2    
    address: 0x44
    update_interval: 10s
  - platform: sht3xd
    temperature:
      name: "sensornode SHT30 0x45 Temperature"
      accuracy_decimals: 2    
    address: 0x45
    update_interval: 10s 
status_led:
  pin:
    number: 12

```

**Logs (if applicable):**

```
[16:59:09][D][api.connection:575]: Client 'Home Assistant 0.95.4 (x.x.x.x)' connected successfully!
[16:59:13][D][sensor:092]: 'SHT30 0x44 Temperature': Sending state 27.32853 °C with 2 decimals of accuracy
[16:59:16][D][sensor:092]: 'SHT30 0x45 Temperature': Sending state 27.15762 °C with 2 decimals of accuracy
[16:59:23][E][i2c:070]: Received NACK on transmit of address 0x44
[16:59:25][E][i2c:070]: Received NACK on transmit of address 0x45

```

**Additional information and things you've tried:**
As you can see from the above log, i've even forced the i2c component to log these events on an ERROR level instead of WARNING, but it did not help.

For information, the sensors begin to work fine after hot-plugging the connector back into the board.

### PR DESCRIPTION
**Pull request by valordk**
## Description:
- Fixed SHT3xd component setting the warning state on sensor disconnection, which allows the Status LED to blink the warning state correctly.
- Added initialization of the SHT3xd sensor to reestablish the connection. (Some other Sensirion sensor models do not require this)

**Related issue (if applicable):** fixes https://github.com/esphome/issues/issues/550

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by glmnet** [comment]
Your changes from https://github.com/esphome/esphome/pull/669 leaked into this PR

**Comment by glmnet** [comment]
There are many complains from Travis code lint, **one** has to do with your code.

**Comment by valordk** [comment]
> 
> 
> There are many complains from Travis code lint, **one** has to do with your code.

Thanks. Should be fixed now. I will recheck other PR and fix the similar issue there.

File: esphome/components/sht3xd/sht3xd.cpp [hunk 1]
```diff
@@ -43,8 +43,14 @@ void SHT3XDComponent::dump_config() {
 }
 float SHT3XDComponent::get_setup_priority() const { return setup_priority::DATA; }
 void SHT3XDComponent::update() {
-  if (!this->write_command_(SHT3XD_COMMAND_POLLING_H))
+  if (this->status_has_warning()) {
+    ESP_LOGD(TAG, "Retrying to reconnect the sensor.");
+    this->write_command_(SHT3XD_COMMAND_SOFT_RESET);
```
Comment (OttoWinter) [inline esphome/components/sht3xd/sht3xd.cpp]: Does the sensor really need a soft reset when something failed? I don't see why the sensor would require this here (no initialization code).

File: esphome/components/sts3x/sensor.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import esphome.codegen as cg
```
Comment (OttoWinter) [inline esphome/components/sts3x/sensor.py]: These changes seem to have snuck into this PR - please see the files tab. Remove these changes from this PR please.

File: esphome/components/sht3xd/sht3xd.cpp (hunk 1 above)
Comment (valordk) [reply esphome/components/sht3xd/sht3xd.cpp]: I suspect that the sensor goes in sleep if not used after reconnection/powerup if not talked to. 
By not calling soft_reset i occasionally got a warning from i2c:
```
Received NACK on transmit of address 0x4B
```

File: esphome/components/sts3x/sensor.py (hunk 2 above)
Comment (valordk) [reply esphome/components/sts3x/sensor.py]: That was a mess. I've cleaned it up, sorry.

**Review by OttoWinter — APPROVED** [review]

**Pull request merged**